src/main/java/de/tum/in/www1/artemis/domain/Team.java

```diff
@@ -0,0 +1,142 @@
+package de.tum.in.www1.artemis.domain;
+
+import java.io.Serializable;
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+
+import javax.persistence.*;
+
+import org.hibernate.annotations.Cache;
+import org.hibernate.annotations.CacheConcurrencyStrategy;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+
+/**
+ * A Team of students.
+ */
+@Entity
+@Table(name = "team")
+@Inheritance(strategy = InheritanceType.SINGLE_TABLE)
+@DiscriminatorColumn(name = "discriminator", discriminatorType = DiscriminatorType.STRING)
+@DiscriminatorValue(value = "T")
+@Cache(usage = CacheConcurrencyStrategy.NONSTRICT_READ_WRITE)
+@JsonInclude(JsonInclude.Include.NON_EMPTY)
+public class Team implements Serializable {
+
+    private static final long serialVersionUID = 1L;
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @Column(name = "name")
+    private String name;
+
+    @Column(name = "image")
+    private String image;
+
+    @ManyToOne
+    private Exercise exercise;
+
+    @ManyToMany
+    @Cache(usage = CacheConcurrencyStrategy.NONSTRICT_READ_WRITE)
+    @JoinTable(name = "team_student", joinColumns = @JoinColumn(name = "team_id", referencedColumnName = "id"), inverseJoinColumns = @JoinColumn(name = "student_id", referencedColumnName = "id"))
+    private Set<User> students = new HashSet<>();
```

Comment:
I think you should make this a bidirectional relation, so that you can easily fetch all teams for a user. So, in `User.java`:
```
@ManyToMany(mappedBy = "students")
@Cache(usage = CacheConcurrencyStrategy.NONSTRICT_READ_WRITE)
private Set<Team> teams;
```

Reply:
I thought about this but I couldn't think of a use case for now where we would want to access all teams for a user, especially across exercises and courses. I think it's enough if I add that direction once the need arises.